assets/js/invidious_iframe_api.js

```diff
@@ -0,0 +1,386 @@
+    seekTo(seconds,allowSeekAhead){//seconds must be a number and allowSeekAhead is ignore
+        seconds = Number(seconds);
+        if(seconds!==NaN&&seconds!==undefined){
+            this.postMessage({eventname:'seek',value:seconds});
+        }
+    }
+    setSize(width,height){//width and height must be Number
+        this.player_iframe.width = Number(width);
+        this.player_iframe.height = Number(height);
+    }
+    getPlaybackRate(){
+        return this.promise_send_event('getplaybackrate');
+    }
+    setPlaybackRate(suggestedRate){//suggestedRate must be number.this player allow not available playback rate such as 1.4
+        suggestedRate = Number(suggestedRate);
+        if(suggestedRate!==NaN&&suggestedRate!==undefined){
+            this.postMessage({eventname:'setplaybackrate',value:suggestedRate});
+        }
+    }
+    getAvailablePlaybackRates(){
+        return this.promise_send_event('getavailableplaybackrates');
+    }
+    playOtherVideoById(option,autoplay,startSeconds_arg){//internal fuction
+        let videoId = '';
+        let startSeconds = -1;
+        let endSeconds = -1;
+        let mediaContetUrl = '';
+        if(typeof option==='string'){
+            if(option.length===11){
+                videoId = option
+            }
+            else{
+                mediaContetUrl = option;
+            }
+            if(startSeconds_arg!==undefined&&typeof startSeconds_arg==='number'){
+                startSeconds = startSeconds_arg;
+            }
+        }
+        else if(typeof option==='object'){
+            if(option.videoId!==undefined&&typeof option.videoId==='string'){
+                if(option.videoId.length==11){
+                    videoId = option.videoId;
+                }
+                else{
+                    this.error_code = 2;
+                    this.event_executor('error');
+                }
+            }
+            else if(option.mediaContentUrl!==undefined&&typeof option.mediaContentUrl==='string'){
+                mediaContetUrl = option.mediaContentUrl;
+            }
+            else{
+                this.error_code = 2;
+                this.event_executor('error');
+            }
+            if(option.startSeconds!==undefined&&typeof option.startSeconds==='number'&&option.startSeconds>=0){
+                startSeconds = option.startSeconds;
+            }
+            if(option.endSeconds!==undefined&&typeof option.endSeconds==='number'&&option.endSeconds>=0){
+                endSeconds = option.endSeconds;
+            }
+        }
+        if(mediaContetUrl.length>0){
+            var tmp_videoId = '';
+            if(mediaContetUrl.indexOf('/v/')!==-1){
+                var end_pos = mediaContetUrl.length-1;
+                if(mediaContetUrl.indexOf('?')!==-1){
+                    end_pos = mediaContetUrl.indexOf('?');
+                }
+                tmp_videoId = mediaContetUrl.substring(mediaContetUrl.indexOf('/v/'),end_pos);
+            }
```

Comment:
Are you trying to catch videoid? This may be done using simple regex like this `\/v\/(\d{11})\??` and using first match group.